sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SchemaPruning.scala

```diff
@@ -48,7 +49,8 @@ object SchemaPruning {
    * right, recursively. That is, left is a "subschema" of right, ignoring order of
    * fields.
    */
-  private def sortLeftFieldsByRight(left: DataType, right: DataType): DataType =
+  private def sortLeftFieldsByRight(left: DataType, right: DataType): DataType = {
```

Comment:
When we construct `mergedDataSchema` in line 39, it's also case-sensitive, isn't it?
> StructType(mergedSchema.filter(f => dataSchemaFieldNames.contains(f.name)))

Reply:
It is. As like `https://github.com/apache/spark/pull/32059#discussion_r608277894`, at `selectField` we treat `GetStructField` and `GetArrayStructFields` differently. So it causes different behavior in case-sensitive aware resolution here.

It looks like we should better correct them together..